draft-rescorla-tls-esni.md

```diff
@@ -179,9 +180,12 @@ structure, defined below.
     } ESNIKeyShareEntry;
 
     struct {
+        uint8 checksum[4];
         ESNIKeyShareEntry keys<4..2^16-1>;
         CipherSuite cipher_suites<2..2^16-2>;
         uint16 padded_length;
+        uint64 not_before;
+        uint64 not_after;
```

Comment:
> a malicious recursive resolver could alter not_before or not_after to prevent a client from using the key as one normally would

I do not think that as an issue, considering the fact that a malicious resolver can do anything including changing the A record or dropping the ESNIKeys entirely.

> It also seems redundant with the TTL field in the RR itself.

The issue is that TTL is always relative to the time when the DNS authoritative server emits the record.

Consider the case where the authoritative server and the TLS server getting out-of-sync.

If we have `not_after`, the servers can be deployed in the a way that it does not cause service disruption. What you need to do setup your system to always generate the ESNIKeys on the TLS server and push it to the authoritative server for distributing it. Then, when the two servers become brain-split, the _esni resource records would become considered as invalid by the clients.

But if we do not have `not_after`, the _esni record that the authoritative server will continue to be considered valid, because TTL is set every time the authoritative server emits the record.

Reply:
Fair points! You've convinced me this is needed.